qiskit_machine_learning/algorithms/effective_dimension.py

```diff
@@ -0,0 +1,337 @@
+            logsum_axis = 1
+        else:
+            n_expanded = n # type: ignore
+            logsum_axis = None
+
+        # calculate effective dimension for each data sample size "n" out
+        # of normalized fishers
+        f_mod = f_hat * n_expanded / (2 * np.pi * np.log(n_expanded))
+        one_plus_fmod = np.eye(self._model.num_weights) + f_mod
+        # take log. of the determinant because of overflow
+        dets = np.linalg.slogdet(one_plus_fmod)[1]
+        # divide by 2 because of square root
+        dets_div = dets / 2
+        effective_dims = (
+            2
+            * (logsumexp(dets_div, axis=logsum_axis) - np.log(self.num_params))
+            / np.log(n / (2 * np.pi * np.log(n)))
+        )
+
+        return np.squeeze(effective_dims)
+
+    def get_eff_dim(self, n: Union[List[int], np.ndarray, int]) -> Union[np.ndarray, int]:
+        """
+        This method compute the effective dimension for a data sample size ``n``.
+
+        Args:
+            n: array of sample sizes
+        Returns:
+             effective_dim: array of effective dimensions for each sample size in n.
+        """
+
+        # pylint: disable=wrong-spelling-in-comment
+        # step 1: Montecarlo sampling
+        grads, output = self.run_montecarlo()
+
+        # step 2: compute as many fisher info. matrices as (input, params) sets
+        fishers = self.get_fishers(gradients=grads, model_outputs=output)
+
+        # step 3: get normalized fisher info matrices
+        f_hat, _ = self.get_fhat(fishers)
+
+        # step 4: compute eff. dim
+        effective_dims = self._get_eff_dim(f_hat, n)
+
+        return effective_dims
+
+
+class LocalEffectiveDimension(EffectiveDimension):
+    """
+    Computes the LOCAL effective dimension for a parametrized model.
+    """
+
+    def __init__(
+        self,
+        qnn: NeuralNetwork,
+        num_inputs: Optional[int] = 1,
+        params: Optional[Union[List, np.ndarray]] = None,
+        inputs: Optional[Union[List, np.ndarray]] = None,
+        fix_seed: bool = False,
+        callback: Optional[Callable] = None,
+    ) -> None:
+        """
+        Args:
+            qnn: A Qiskit NeuralNetwork, with a specific number of weights
+                (qnn_num_weights) and input size (qnn_input_size)
+            num_inputs:  Number of inputs, if user chooses to randomly sample
+                from a normal distribution.
+            params: An array of neural network weights, of shape (1, qnn_num_weights).
+            inputs: An array of inputs to the neural network, of shape
+                (num_inputs, qnn_input_size).
+
+        Raises:
+            QiskitMachineLearningError: If ``len(params) > 1``
+        """
+        params = np.asarray(params)
+        if params is not None and len(params.shape) > 1 and params.shape[0] > 1:
+            raise QiskitMachineLearningError(
+                "The local effective dimension algorithm uses only 1 set of parameters."
+            )
```

Comment:
It is helpful to include for uses and when debugging etc to include specific information about the failure in a formatted string in the message.

Though params is checked here there is a setter in the parent right so that this check would not be done later if you call the setter. Perhaps the setter should be overidden here and make the check, since params passed in the constructor is set using the setter it would protect both paths.